src/App.tsx

```diff
@@ -1,61 +1,148 @@
 import './App.scss';
+import { ChangeEvent, FC, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(id: number):User | null {
+  const foundUser = usersFromServer.find(user => user.id === id);
+
+  return foundUser || null;
+}
+
+function finderBiggestId(todos: Todo[]): number {
+  return Math.max(...todos.map(todo => todo.id));
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: FC = () => {
+  const [visibleTodos, setTodos] = useState(todos);
+  const [isSelectedUserError, setSelectionError] = useState(false);
+  const [isTitleError, setTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const handlerTitleChange = (
+    event: ChangeEvent<HTMLInputElement>,
+  ) => {
+    const formattedTitle = event.target.value.replace(
+      /[^a-zA-Zа-яА-Я0-9\s]/g, '',
+    );
+
+    if (formattedTitle) {
+      setTitle(formattedTitle);
+      setTitleError(false);
+    }
+  };
+
+  const submitHandler = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+
+    const biggerId = finderBiggestId(visibleTodos);
+    const exactlyUser = getUserById(userId);
+    const formattedTitle = title.trim();
+
+    if (!formattedTitle) {
+      setTitleError(true);
+    }
+
+    if (!exactlyUser) {
+      setSelectionError(true);
+    }
+
+    if (!formattedTitle || !exactlyUser) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: biggerId + 1,
+      title: formattedTitle,
+      completed: false,
+      userId: exactlyUser.id,
+      user: exactlyUser,
+    };
+
+    setTodos([...visibleTodos, newTodo]);
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSelectionChange = (event: { target: { value: string; }; }) => {
```

Comment:
Change the parameter type of 'handleSelectionChange' to 'React.ChangeEvent<HTMLSelectElement>' for better type handling.